File: eduarda/pset5/roundcorridor.cpp
#include <bits/stdc++.h>
using namespace std;


int main(){
    long long n, m, mdc, k[3];
    int q;
    cin >> n >> m >> q;
    k[3], mdc = __gcd(n, m);
    k[1] = n/mdc;
    k[2] = m/mdc;
    while(q--){
        long long sx, sy, ex, ey;
        cin >> sx >> sy >> ex >> ey;
        long long a1 = sy/k[sx] + (sy%k[sx] != 0);
        long long a2 = ey/k[ex] + (ey%k[ex] != 0);
        if(a1 == a2) cout << "YES" << endl;
        else cout << "NO" << endl;
    }
}

File: eduarda/pset5/roundcorridor.py
import math

def gcd(a, b):
    if b == 0:
        return a
    return gcd(b, a % b)

n, m, q = map(int, input().split())

for _ in range(q):
    sx, sy, ex, ey = map(int, input().split())

    if sx == ex:
        if sy == ey:
            print("NO")
        else:
            print("YES")
    else:
        if gcd(n, m) == 1 or gcd(sy, ey) == 1:
            print("YES")
        else:
            print("NO")